app/jobs/send_notification_job.rb

```diff
@@ -2,11 +2,32 @@
 
 # Sends a simple one-step notification.
 class SendNotificationJob < ApplicationJob
+  def self.disable_endpoint(endpoint)
+    endpoint.active = false
+    endpoint.save!
+  end
+
+  discard_on(Notifications::FatalError) do |job, error|
+    endpoint = job.endpoint
+    disable_endpoint(endpoint)
+    Rails.logger.warn("Disabled #{endpoint} after receiving #{error.inspect}")
+  end
+
+  retry_on(Notifications::RecoverableError, wait: :exponentially_longer, attempts: 3) do |job, error|
+    endpoint = job.endpoint
+    disable_endpoint(endpoint)
+    Rails.logger.warn("Disabled #{endpoint} after too many retries for #{error.inspect}")
+  end
+
   def perform(endpoint, timestamp, category, message)
     Rails.logger.debug("Received a job to send: #{endpoint}, #{timestamp}, #{category}, #{message}")
 
     endpoint.send_message(timestamp: timestamp, category: category, message: message)
```

Comment:
If a delivery of message A gets queued, endpoint gets disabled then A gets processed, we would still try to deliver A even though the endpoint is disabled. Maybe we should check if the endpoint is active before hitting this?

Reply:
I will add a condition to discard messages if the endpoint is not active